components/scrubber/scrubber.go

```diff
@@ -152,14 +159,39 @@ func (s *scrubberImpl) KeyValue(key string, value string) (sanitisedValue string
 	return sanitisatiser(value)
 }
 
+type keySanitiser struct {
+	s Sanitisatiser
+}
+
+var (
+	sanitiseIgnore keySanitiser = keySanitiser{s: nil}
+	sanitiseHash   keySanitiser = keySanitiser{s: SanitiseHash}
+	sanitiseRedact keySanitiser = keySanitiser{s: SanitiseRedact}
+)
+
 // getSanitisatiser implements
 func (s *scrubberImpl) getSanitisatiser(key string) Sanitisatiser {
-	for re, sanitisatiser := range s.RegexpIndex {
-		if re.MatchString(key) {
-			return sanitisatiser
+	lower := strings.ToLower(key)
+	san, ok := s.KeySanitiserCache.Get(lower)
+	if ok {
+		w := san.(keySanitiser)
+		return w.s
+	}
+
+	for _, f := range s.LowerSanitiseRedact {
+		if strings.Contains(lower, f) {
+			s.KeySanitiserCache.Add(lower, sanitiseRedact)
+			return SanitiseRedact
+		}
+	}
+	for _, f := range s.LowerSanitiseHash {
+		if strings.Contains(lower, f) {
+			s.KeySanitiserCache.Add(lower, sanitiseHash)
+			return SanitiseHash
 		}
 	}
 
+	s.KeySanitiserCache.Add(lower, sanitiseIgnore)
```

Comment:
this can grow fast

Reply:
the lru cache has a max size of 1000, I also think that keys won't be too unique and have a high chance of cache hits (e.g. log field names or metric label names are mostly constant). For a 1MB metrics file I observed there's only ~300 unique keys

Reply:
let's try